### PR TITLE
[Core] replacementprocess in submodelparts

### PR DESCRIPTION
**Pull request by philbucher**
with those modifications it is possible to:
- replace elems/conds in only a submodelpart (default does not change, everything is replaced)
- only replace elements OR conditions

test is added

requested by @inigolcanalejo

### CODE REVIEW

File: kratos/processes/replace_elements_and_condition_process.cpp [hunk 1]
```diff
@@ -27,81 +27,88 @@ namespace Kratos
...
+        replace_conditions = true;
+        const Condition& rReferenceCondition = KratosComponents<Condition>::Get(mSettings["condition_name"].GetString());
 
-/***********************************************************************************/
-/***********************************************************************************/
+        #pragma omp parallel for
+        for(int i=0; i< static_cast<int>(r_model_part.Conditions().size()); i++) {
+            auto it_cond = r_model_part.ConditionsBegin() + i;
 
-ModelPart& ReplaceElementsAndConditionsProcess::ObtainRootModelPart( ModelPart& rModelPart )
```
Comment (philbucher) [inline kratos/processes/replace_elements_and_condition_process.cpp:71]: this method already exists in the modelpart

File: kratos/tests/cpp_tests/processes/test_replace_elements_and_condition_process.cpp [hunk 2]
```diff
@@ -237,23 +237,97 @@ namespace Kratos
...
+            ReplaceElementsAndConditionsProcess process = ReplaceElementsAndConditionsProcess(this_sub_model_part, settings);
+            process.Execute();
+
+            // Same element type, same geometry type
+            std::string component_name;
+            for (auto& r_element : this_sub_model_part.Elements()) {
+                CompareElementsAndConditionsUtility::GetRegisteredName(r_element, component_name);
+                KRATOS_CHECK_EQUAL(component_name, "Element2D3N");
+            }
+        }
```
Comment (philbucher) [inline kratos/tests/cpp_tests/processes/test_replace_elements_and_condition_process.cpp:331]: I don't know how to test the elements that were not replaced
Unfortuanately the same way of checking does not work if the elements are baseclass-elements 
Also there are no other elements in the core that I could use

Does anyone have an idea?

File: kratos/processes/replace_elements_and_condition_process.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/processes/replace_elements_and_condition_process.cpp:71]: Yes..., sorry for that

File: kratos/tests/cpp_tests/processes/test_replace_elements_and_condition_process.cpp (hunk 2 above)
Comment (loumalouomega) [reply kratos/tests/cpp_tests/processes/test_replace_elements_and_condition_process.cpp:331]: Compare elements is possible, but is a little bit of work, look at the model_part_io.h when the name of the output element is checked it does a loop over the KratosComponents<Element> and KratosComponents<Condition>

**Review by loumalouomega — APPROVED** [review]
Looks OK for me, thanks

File: kratos/tests/cpp_tests/processes/test_replace_elements_and_condition_process.cpp (hunk 2 above)
Comment (philbucher) [reply kratos/tests/cpp_tests/processes/test_replace_elements_and_condition_process.cpp:331]: comparing elements yes, but the elements in the MainModelpart (i.e. the ones that are not replaced) are Baseclass-elements => the `CompareElementsAndConditionsUtility` fails here :/

**Comment by philbucher** [comment]
closing since this does not work properly

I.e. if the elements we replace in a submodelpart are also contained in another submodelpart, then those won't be replaced
We couldn't come up with a good solution, hence this PR does not make sense

=> @inigolcanalejo already came up with a solution for his (specific) problem

**Comment by inigolcanalejo** [comment]
Maybe not all of the changes are necessary, but I think you did some cleaning that made sense.

**Pull request closed without merging**